tests/core/test_output_nft.c

```diff
@@ -0,0 +1,452 @@
+  TEST_ASSERT(output_nft_serialize(output, output_nft_buf, 1) == 0);  // expect serialization fails
+  TEST_ASSERT(output_nft_serialize(output, output_nft_buf, output_nft_expected_len) == output_nft_expected_len);
+
+  // deserialize NFT Output and validate it
+  output_nft_t* deser_output = output_nft_deserialize(output_nft_buf, 1);
+  TEST_ASSERT_NULL(deser_output);  // expect deserialization fails
+  deser_output = output_nft_deserialize(output_nft_buf, output_nft_expected_len);
+  TEST_ASSERT_NOT_NULL(deser_output);
+  TEST_ASSERT_EQUAL_UINT8(ADDRESS_TYPE_NFT, deser_output->address->type);
+  TEST_ASSERT_EQUAL_MEMORY(deser_output->address, &addr, 1 + ADDRESS_NFT_BYTES);
+  TEST_ASSERT_EQUAL_UINT64(123456789, deser_output->amount);
+
+  TEST_ASSERT_NOT_NULL(deser_output->native_tokens);
+  TEST_ASSERT_EQUAL_UINT32(3, native_tokens_count(&deser_output->native_tokens));
+  token = deser_output->native_tokens;
+  TEST_ASSERT_EQUAL_MEMORY(token_id1, token->token_id, NATIVE_TOKEN_ID_BYTES);
+  TEST_ASSERT_EQUAL_MEMORY(amount1, token->amount, sizeof(uint256_t));
+  token = token->hh.next;
+  TEST_ASSERT_EQUAL_MEMORY(token_id2, token->token_id, NATIVE_TOKEN_ID_BYTES);
+  TEST_ASSERT_EQUAL_MEMORY(amount2, token->amount, sizeof(uint256_t));
+  token = token->hh.next;
+  TEST_ASSERT_EQUAL_MEMORY(token_id3, token->token_id, NATIVE_TOKEN_ID_BYTES);
+  TEST_ASSERT_EQUAL_MEMORY(amount3, token->amount, sizeof(uint256_t));
+
+  TEST_ASSERT_EQUAL_MEMORY(nft_id, deser_output->nft_id, ADDRESS_NFT_BYTES);
+  TEST_ASSERT_EQUAL_INT32(sizeof("Test metadata..."), deser_output->immutable_metadata->len);
+  TEST_ASSERT_EQUAL_MEMORY("Test metadata...", deser_output->immutable_metadata->data,
+                           deser_output->immutable_metadata->len);
+
+  TEST_ASSERT_NOT_NULL(deser_output->feature_blocks);
+  feat_block = feat_blk_list_get(deser_output->feature_blocks, 0);
+  TEST_ASSERT_EQUAL_UINT8(FEAT_SENDER_BLOCK, feat_block->type);
+  TEST_ASSERT_EQUAL_MEMORY(&addr, feat_block->block, ADDRESS_NFT_BYTES);
+  feat_block = feat_blk_list_get(deser_output->feature_blocks, 1);
+  TEST_ASSERT_EQUAL_UINT8(FEAT_DUST_DEP_RET_BLOCK, feat_block->type);
+  TEST_ASSERT_EQUAL_UINT64(1000000, *((uint64_t*)feat_block->block));
+
+  output_nft_print(output);
+
+  // clean up
+  free(amount1);
+  free(amount2);
+  free(amount3);
+  free(output_nft_buf);
+  byte_buf_free(metadata);
+  native_tokens_free(&native_tokens);
+  free_feat_blk_list(feat_blocks);
+  output_nft_free(output);
+  output_nft_free(deser_output);
+}
+
+void test_output_nft_without_native_tokens() {
+  // create random NFT address
+  address_t addr = {};
+  addr.type = ADDRESS_TYPE_NFT;
+  iota_crypto_randombytes(addr.address, ADDRESS_NFT_BYTES);
+
+  // create NFT ID
+  byte_t nft_id[ADDRESS_NFT_BYTES];
+  memcpy(nft_id, addr.address, ADDRESS_NFT_BYTES);
+
+  // create metadata
+  byte_buf_t* metadata = byte_buf_new_with_data("Test metadata...", sizeof("Test metadata..."));
```

Comment:
please fix the following warnings
```
[build] ../tests/core/test_output_nft.c:41:49: warning: passing 'char [17]' to parameter of type 'byte_t *' (aka 'unsigned char *') converts between pointers to integer types with different sign [-Wpointer-sign]
[build]   byte_buf_t* metadata = byte_buf_new_with_data("Test metadata...", sizeof("Test metadata..."));
[build]                                                 ^~~~~~~~~~~~~~~~~~
[build] ../src/core/utils/byte_buffer.h:40:43: note: passing argument to parameter 'data' here
[build] byte_buf_t* byte_buf_new_with_data(byte_t data[], size_t len);
[build]                                           ^
[build] ../tests/core/test_output_nft.c:150:49: warning: passing 'char [17]' to parameter of type 'byte_t *' (aka 'unsigned char *') converts between pointers to integer types with different sign [-Wpointer-sign]
[build]   byte_buf_t* metadata = byte_buf_new_with_data("Test metadata...", sizeof("Test metadata..."));
[build]                                                 ^~~~~~~~~~~~~~~~~~
[build] ../src/core/utils/byte_buffer.h:40:43: note: passing argument to parameter 'data' here
[build] byte_buf_t* byte_buf_new_with_data(byte_t data[], size_t len);
[build]                                           ^
[build] ../tests/core/test_output_nft.c:365:49: warning: passing 'char [17]' to parameter of type 'byte_t *' (aka 'unsigned char *') converts between pointers to integer types with different sign [-Wpointer-sign]
[build]   byte_buf_t* metadata = byte_buf_new_with_data("Test metadata...", sizeof("Test metadata..."));
[build]                                                 ^~~~~~~~~~~~~~~~~~
[build] ../src/core/utils/byte_buffer.h:40:43: note: passing argument to parameter 'data' here
[build] byte_buf_t* byte_buf_new_with_data(byte_t data[], size_t len);
[build]                                           ^
[build] 3 warnings generated.
```

Reply:
I didn't check compilation with Clang but only with GCC. For the future, both are needed to be used before creating PR.